server/commands/check.go

```diff
@@ -49,12 +49,11 @@ func NewCheckQuery(datastore storage.OpenFGADatastore, t trace.Tracer, m metric.
 
 // Execute the query in `checkRequest`, returning the response or an error.
 func (query *CheckQuery) Execute(ctx context.Context, req *openfgapb.CheckRequest) (*openfgapb.CheckResponse, error) {
-	statCheckResolutionDepth, _ := query.meter.AsyncInt64().Gauge(
+	statCheckResolutionDepth, _ := query.meter.SyncInt64().Counter(
```

Comment:
What is the reasoning for this change? 


 See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter

> Asynchronous Counter is an [asynchronous Instrument](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-instrument-api) which reports [monotonically](https://wikipedia.org/wiki/Monotonic_function) increasing value(s) when the instrument is being observed. Example uses for Asynchronous Counter: [CPU time](https://wikipedia.org/wiki/CPU_time)

Resolution depth is not monotonically increasing, it varies per request.

Reply:
For an unknown reason, I couldn't manage this metric to be working when using asynchronous counter. I'll try to dig into it more in the coming days

Reply:
I tried to investigate why I did this: 
- Gauge has been changed to counter because it is not available in sync mode, but counter doesn't fit here as you mentioned; 
- With an async method, I not sure why, but the gauge doesn't register any metric in this context

Even though it would be working, would that make sense to keep a gauge here ? As its values would change too frequently to be accurate or to have an average depth of queries.

Reply:
I had a bit of a read and am now convinced that Gauge is correct here. As you say, there is no sync method now, but maybe there will be one soon 🤞 https://github.com/open-telemetry/opentelemetry-specification/issues/2318

I think it probably doesn't work as we are not registering the instrument in a callback. The docs for `AsyncInt64()` say
> 	// To Observe data with instruments it must be registered in a callback.

I don't know, but do we have to do something like the following instead of just calling `Observe` like we are now? 
```
_ = query.meter.RegisterCallback([]instrument.Asynchronous{statCheckResolutionDepth}, func(ctx context.Context) {
	statCheckResolutionDepth.Observe(ctx, int64(rc.metadata.GetResolve()))
})
```

Maybe it makes more sense to just remove this instrument for now until there is a sync gauge?